### DISCUSSION TITLE
ProxBox - Server Error

### DISCUSSION THREAD

**Discussion by niallmcpartland** [netbox-community/netbox#15108] [Help Wanted!] [answered]
Hello,
I have just setup ProxBox however i am recieving the following error.
It isnt really pointing me in the right direction and to be honest i have no idea on where to start to diagnose. 

`The request failed with code 400 Bad Request: {'disk': ['The specified disk size (63) must match the aggregate size of assigned virtual disks (60).']}`

**Comment by candlerb** [comment, marked as answer]
Do you mean https://github.com/netdevopsbr/netbox-proxbox ?

That's a third-party piece of software. You can see a list of contact points for getting help with it here:
https://github.com/netdevopsbr/netbox-proxbox?tab=readme-ov-file#8-get-help-from-community

I do note that the error message you see is returned by Netbox, see source code [here](https://github.com/netbox-community/netbox/blob/v3.7.2/netbox/virtualization/models/virtualmachines.py#L208-L209):

```
        # Validate aggregate disk size
        if self.pk:
            total_disk = self.virtualdisks.aggregate(Sum('size', default=0))['size__sum']
            if total_disk and self.disk is None:
                self.disk = total_disk
            elif total_disk and self.disk != total_disk:
                raise ValidationError({
                    'disk': _(
                        "The specified disk size ({size}) must match the aggregate size of assigned virtual disks "
                        "({total_size})."
                    ).format(size=self.disk, total_size=total_disk)
                })
```

Virtual machine disks were a new feature in [v3.7.0](https://github.com/netbox-community/netbox/releases/v3.7.0). You can see from this code that if you specify individual disks, their size must add up to the VM's total disk size. If netbox-proxbox isn't doing that calculation correctly, then you should raise it as an issue with that project (including steps to reproduce, i.e. what disk sizes your VM has). Maybe it's a [GB/GiB](https://en.wikipedia.org/wiki/Gigabyte) problem.
